### ISSUE TITLE
My model has stopped improving please can you help

### ISSUE THREAD

**Issue by KimberleyJensen** [facebookresearch/demucs#198]
Hello I have just reached epoch 100 for my model but it doesnt really seem to be improving anymore, i am training it with 91 tracks from musdb using this command

python3.8 run.py --wav /demucs/musdbe --musdb /demucs/musdbe --is_wav --channels=48 --depth=6 --repitch=0 --samplerate 44100 --samples 44100 --data_stride 44100 -w 8

I submitted some of my models to the mdx challenge but my highest score is only 4.1, please could you tell me if any of my settings are wrong. Also if its ok please will you share your training and validation loss numbers with me for your models that are just trained on musdb.

here is the training and validation numbers

Epoch 000: train=0.17936380 valid=0.23429446 best=0.2343 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 001: train=0.16199168 valid=0.22592303 best=0.2259 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 002: train=0.15478575 valid=0.21981589 best=0.2198 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 003: train=0.14943420 valid=0.21598954 best=0.2160 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 004: train=0.14530816 valid=0.21383869 best=0.2138 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 005: train=0.14147505 valid=0.21538643 best=0.2138 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 006: train=0.13927522 valid=0.21655120 best=0.2138 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 007: train=0.13661285 valid=0.21057392 best=0.2106 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 008: train=0.13476804 valid=0.21411839 best=0.2106 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 009: train=0.13325076 valid=0.20646511 best=0.2065 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 010: train=0.12992045 valid=0.20981736 best=0.2065 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 011: train=0.13014731 valid=0.21011194 best=0.2065 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 012: train=0.12841646 valid=0.20465061 best=0.2047 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 013: train=0.12631190 valid=0.20728803 best=0.2047 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 014: train=0.12557334 valid=0.20157171 best=0.2016 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 015: train=0.12436709 valid=0.20831249 best=0.2016 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 016: train=0.12276602 valid=0.19952788 best=0.1995 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 017: train=0.12190687 valid=0.20284905 best=0.1995 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 018: train=0.12144498 valid=0.20005746 best=0.1995 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 019: train=0.11935077 valid=0.20036335 best=0.1995 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 020: train=0.11948095 valid=0.19797493 best=0.1980 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 021: train=0.11804976 valid=0.19565350 best=0.1957 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 022: train=0.11731628 valid=0.19515904 best=0.1952 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 023: train=0.11721382 valid=0.19851315 best=0.1952 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 024: train=0.11581732 valid=0.19655280 best=0.1952 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 025: train=0.11549132 valid=0.19627299 best=0.1952 ms=0.00MB cms=0.00MB duration=0.51 hrs
Epoch 026: train=0.11474334 valid=0.19577430 best=0.1952 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 027: train=0.11422826 valid=0.19321651 best=0.1932 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 028: train=0.11326355 valid=0.19831502 best=0.1932 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 029: train=0.11243353 valid=0.19226982 best=0.1923 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 030: train=0.11144218 valid=0.19571391 best=0.1923 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 031: train=0.11093600 valid=0.19432431 best=0.1923 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 032: train=0.11044605 valid=0.19317250 best=0.1923 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 033: train=0.11005729 valid=0.19316754 best=0.1923 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 034: train=0.11066689 valid=0.19126716 best=0.1913 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 035: train=0.10879415 valid=0.19530303 best=0.1913 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 036: train=0.10907283 valid=0.19135283 best=0.1913 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 037: train=0.10799820 valid=0.19122263 best=0.1912 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 038: train=0.10793418 valid=0.19047101 best=0.1905 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 039: train=0.10774017 valid=0.19217015 best=0.1905 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 040: train=0.10663013 valid=0.19213865 best=0.1905 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 041: train=0.10671654 valid=0.18842015 best=0.1884 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 042: train=0.10669713 valid=0.18868685 best=0.1884 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 043: train=0.10586745 valid=0.18773609 best=0.1877 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 044: train=0.10501046 valid=0.19060865 best=0.1877 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 045: train=0.10521723 valid=0.18864295 best=0.1877 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 046: train=0.10450020 valid=0.18779382 best=0.1877 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 047: train=0.10392914 valid=0.19244286 best=0.1877 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 048: train=0.10459584 valid=0.18813983 best=0.1877 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 049: train=0.10354572 valid=0.19126991 best=0.1877 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 050: train=0.10368152 valid=0.19067149 best=0.1877 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 051: train=0.10350154 valid=0.18990243 best=0.1877 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 052: train=0.10265200 valid=0.18607843 best=0.1861 ms=0.00MB cms=0.00MB duration=0.52 hrs
Epoch 053: train=0.10222826 valid=0.18970347 best=0.1861 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 054: train=0.10222677 valid=0.19015039 best=0.1861 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 055: train=0.10149915 valid=0.18674213 best=0.1861 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 056: train=0.10147599 valid=0.18636963 best=0.1861 ms=0.00MB cms=0.00MB duration=0.51 hrs
Epoch 057: train=0.10068434 valid=0.18780379 best=0.1861 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 058: train=0.10074593 valid=0.18696774 best=0.1861 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 059: train=0.10042134 valid=0.18728969 best=0.1861 ms=0.00MB cms=0.00MB duration=0.51 hrs
Epoch 060: train=0.09994880 valid=0.18745616 best=0.1861 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 061: train=0.10009010 valid=0.18561919 best=0.1856 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 062: train=0.09893880 valid=0.18791920 best=0.1856 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 063: train=0.09987734 valid=0.18557964 best=0.1856 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 064: train=0.09892482 valid=0.18694821 best=0.1856 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 065: train=0.09858215 valid=0.18787349 best=0.1856 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 066: train=0.09851859 valid=0.18679359 best=0.1856 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 067: train=0.09850492 valid=0.18673794 best=0.1856 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 068: train=0.09833616 valid=0.18672953 best=0.1856 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 069: train=0.09809921 valid=0.18642373 best=0.1856 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 070: train=0.09771808 valid=0.18456551 best=0.1846 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 071: train=0.09719046 valid=0.18858263 best=0.1846 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 072: train=0.09709723 valid=0.18646447 best=0.1846 ms=0.00MB cms=0.00MB duration=0.53 hrs
Epoch 073: train=0.09670861 valid=0.18577690 best=0.1846 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 074: train=0.09650598 valid=0.18579219 best=0.1846 ms=0.00MB cms=0.00MB duration=0.51 hrs
Epoch 075: train=0.09700923 valid=0.18458719 best=0.1846 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 076: train=0.09575461 valid=0.18320294 best=0.1832 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 077: train=0.09543572 valid=0.18388865 best=0.1832 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 078: train=0.09532938 valid=0.18639356 best=0.1832 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 079: train=0.09500796 valid=0.18521578 best=0.1832 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 080: train=0.09483501 valid=0.18578621 best=0.1832 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 081: train=0.09430454 valid=0.18547933 best=0.1832 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 082: train=0.09285709 valid=0.18599143 best=0.1832 ms=0.00MB cms=0.00MB duration=0.52 hrs
Epoch 083: train=0.09318674 valid=0.18546987 best=0.1832 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 084: train=0.09285351 valid=0.18384251 best=0.1832 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 085: train=0.09197937 valid=0.18376233 best=0.1832 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 086: train=0.09219379 valid=0.18408341 best=0.1832 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 087: train=0.09146286 valid=0.18624981 best=0.1832 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 088: train=0.09160108 valid=0.18769440 best=0.1832 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 089: train=0.09116302 valid=0.18342389 best=0.1832 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 090: train=0.09091037 valid=0.18759798 best=0.1832 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 091: train=0.09443546 valid=0.18473786 best=0.1832 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 092: train=0.09303001 valid=0.18774036 best=0.1832 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 093: train=0.09352115 valid=0.18400948 best=0.1832 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 094: train=0.09322173 valid=0.18368874 best=0.1832 ms=0.00MB cms=0.00MB duration=0.48 hrs
Epoch 095: train=0.09276818 valid=0.18229246 best=0.1823 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 096: train=0.09271044 valid=0.18486367 best=0.1823 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 097: train=0.09269471 valid=0.18366929 best=0.1823 ms=0.00MB cms=0.00MB duration=0.50 hrs
Epoch 098: train=0.09238645 valid=0.18192695 best=0.1823 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 099: train=0.09243721 valid=0.18288598 best=0.1823 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 100: train=0.09218158 valid=0.18300913 best=0.1823 ms=0.00MB cms=0.00MB duration=0.49 hrs
Epoch 101: train=0.09138964 valid=0.18352183 best=0.1823 ms=0.00MB cms=0.00MB duration=0.50 hrs

Thank you for all your help !!

**Comment by adefossez** [comment]
The segment length on which you are training is way too small (44100 is 1 second, while in the Demucs paper I train on 10 seconds). This is likely why your model stop progressing. If I were you I would reduce the batch size but increase the number of samples. You need at least 5 to 6 seconds.
